jbpm-case-mgmt/jbpm-case-mgmt-impl/src/test/java/org/jbpm/casemgmt/impl/StageCompletionConditionTest.java

```diff
@@ -0,0 +1,359 @@
+        assertCaseInstanceNotActive(caseId);
+    }
+
+    @Test
+    public void testProcessVariable() {
+        Map<String, Object> data = new HashMap<>();
+        data.put("continue", false);
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(),
+                STAGE_WITH_TASK_PROCESS_VARIABLE, data, Collections.emptyMap());
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), STAGE_WITH_TASK_PROCESS_VARIABLE, caseFile);
+        assertThat(caseId).isNotNull().isEqualTo(FIRST_CASE_ID);
+        assertCaseInstanceActive(caseId);
+
+        CaseInstance caseInstance = caseService.getCaseInstance(caseId);
+        assertThat(caseInstance).isNotNull();
+        Long processInstanceId = ((CaseInstanceImpl) caseInstance).getProcessInstanceId();
+        processService.setProcessVariable(processInstanceId, "continue", false);
+
+        caseService.addDataToCaseFile(caseId, "continue", true);
+        assertCaseInstanceActive(caseId);
+
+        processService.setProcessVariable(processInstanceId, "continue", true);
+        assertCaseInstanceNotActive(caseId);
+    }
+
+    @Test
+    public void testCaseFileVariable() {
+        Map<String, Object> data = new HashMap<>();
+        data.put("continue", false);
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(),
+                STAGE_WITH_TASK_CASE_FILE_VARIABLE, data, Collections.emptyMap());
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), STAGE_WITH_TASK_CASE_FILE_VARIABLE, caseFile);
+        assertThat(caseId).isNotNull().isEqualTo(FIRST_CASE_ID);
+        assertCaseInstanceActive(caseId);
+
+        caseService.addDataToCaseFile(caseId, "continue", true);
+        assertCaseInstanceNotActive(caseId);
+    }
+
+    @Test
+    public void testCaseFileVariableLastTask() {
+        Map<String, Object> data = new HashMap<>();
+        data.put("continue", false);
+        CaseFileInstance caseFile = caseService.newCaseFileInstance(deploymentUnit.getIdentifier(),
+                STAGE_WITH_TASK_CASE_FILE_VARIABLE, data, Collections.emptyMap());
+
+        String caseId = caseService.startCase(deploymentUnit.getIdentifier(), STAGE_WITH_TASK_CASE_FILE_VARIABLE, caseFile);
+        assertThat(caseId).isNotNull().isEqualTo(FIRST_CASE_ID);
+        assertCaseInstanceActive(caseId);
+
+        List<TaskSummary> tasks = runtimeDataService.getTasksAssignedAsPotentialOwner(USER, new QueryFilter());
+        assertThat(tasks).hasSize(1);
+        TaskSummary insideTask = tasks.get(0);
+        assertThat(insideTask.getName()).isEqualTo(INSIDE_TASK);
+        userTaskService.completeAutoProgress(insideTask.getId(), USER, null);
+        assertCaseInstanceActive(caseId);
+
+        caseService.addDataToCaseFile(caseId, "continue", true);
+        assertCaseInstanceNotActive(caseId);
+    }
+
+    @Test
+    @Ignore
```

Comment:
It would be good to tell the reason for ignoring a test - @Ignore annotation has a parameter for that.